tests/subsys/net/lib/nrf_cloud/src/main.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/nrf_cloud.h>
+#include "nrf_cloud_fsm.h"
+
+/* Main test function that declares a test suite
+*  includes all the test cases implemented below
+*/
+ZTEST_SUITE(nrf_cloud_init_test, NULL, NULL, NULL, NULL, NULL);
+
+/* The library can only be initialized when the internal state is
+*  not idle, so the test verifies that nrf_loud_init function will
+*  return with permission denied error (-EACCES) with said condition
+*/
+ZTEST(nrf_cloud_init_test, test_init_not_idle)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = NULL,
+		.client_id = NULL,
+	};
+
+	/* The function is an interface (declared in nrf_cloud_fsm.h)
+	*  and implemented inside nrf_cloud.c, so mocking would not be
+	*  a good idea. Here it changes the internal state to not idle
+	*/
+	nfsm_set_current_state_and_notify(STATE_TOTAL, NULL);
+	zassert_equal(-EACCES, nrf_cloud_init(&params),
+			  "return should be -EACCES when not idle");
+}
+
+/* Verifiy that nrf_cloud_init function will throw an error
+*  of invalid argument (-EINVAL) with a null parameter
+*/
+ZTEST(nrf_cloud_init_test, test_init_null_params)
+{
+	nfsm_set_current_state_and_notify(STATE_IDLE, NULL);
+	zassert_equal(-EINVAL, nrf_cloud_init(NULL),
+			  "return should be -EINVAL with NULL params");
+}
+
+/* Verifiy that nrf_cloud_init function will throw an error
+*  of invalid argument (-EINVAL) when a null event handler
+*  is passed in as part of its parameters
+*/
+ZTEST(nrf_cloud_init_test, test_init_null_handler)
+{
+	struct nrf_cloud_init_param params = {
+		.event_handler = NULL,
+		.client_id = NULL,
+	};
+
+	/* The state is changed to idle */
+	nfsm_set_current_state_and_notify(STATE_IDLE, NULL);
+	zassert_equal(-EINVAL, nrf_cloud_init(&params),
+			  "return should be -EINVAL with NULL event handler");
+}
+
+/* Simple event handler needed for the init function to
+*  have a success initialization in the test below
+*/
+void event_handler(const struct nrf_cloud_evt *const evt)
+{
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_READY:
+		printk("Device is connected and ready");
+		break;
+	case NRF_CLOUD_EVT_ERROR:
+		printk("Failed to connect to cloud");
+		break;
+	default:
+		break;
+	}
+}
+
+/* Verify that nrf_cloud_init function will be success with
+*  the right internal state and a valid event handler
+*/
```

Comment:
Nit: Slight rephrasing might work well here:
```suggestion
/* Verify that nrf_cloud_init function will succeed given
*  valid internal state and event handler
*/
```